### PR TITLE
Add test for Content API filter by locale

### LINKED ISSUE #11953: Cannot get single type in the selected locale
### Describe the bug

When multiple locales are set in Strapi, fetching a single type in the selected locale is not working properly. It always returns a response in the default locale. 
Moreover, it's not clear whether the provided steps in the [i18n setup](https://docs.strapi.io/developer-docs/latest/plugins/i18n.html) are also valid for single types.

### Steps to reproduce the behavior

1. Go to the admin panel and add a new locale (e.g. German) via Settings - Internationalization. Save the changes.
2. Create a simple single type in Content-Type Builder, e.g. for the about section let's call it "About". In Advanced settings, enable the localization by checking the option "Enable localization for this Content-Type".
3. Create a long text field element within "About" single type and name it e.g. "summary". Save the changes.
4. In Content Manager, fill the "summary" field with some text and click on save + publish for both locales.
5. Update the permissions to read the content type. Go to Settings - under section Roles and permissions plugin, click on Roles - Public - About - check the box "find" and save the changes.
6. Re-run the application and try to get the German locale with `http://localhost:1337/api/about?_locale=de` via Postman or Thunder Client.
7. It will get a response in default locale, instead of German (see the 1st screenshot).

### Expected behavior

It should return a response in the selected locale.

### Screenshots

Screenshot 2021-12-20 at 14 23 16

Screenshot 2021-12-20 at 14 56 17

### System

- Node.js version: `v14.18.1`
- NPM version: `v6.14.15`
- Strapi version: `v4.0.0`
- Database: `sqlite3: v5.0.2`
- Operating system: `macOS Monterey v12.1`

### Additional context

I noticed that when trying to get all available locales for that single type with `http://localhost:1337/api/about?_locale=all` it will receive only the content with the default locale (see 2nd screenshot).

Any help would be appreciated!

### PR DESCRIPTION
**Pull request by iicdii**
Signed-off-by: harimkims <harimkims@gmail.com>

### What does it do?

- Add e2e tests for filtering by locale (collection-type/single-type)

### Why is it needed?

~- To enable filter by locale~

~Following [the documentation](https://docs.strapi.io/developer-docs/latest/plugins/i18n.html#getting-localized-entries-with-the-locale-parameter), you can filter content by `_locale` parameter but the identifier was `locale` in code.~

It turns out the documentation didn't reflect the change the usage of `_locale` becomes `locale` in i18n api.
So I revert changes and just keep the e2e tests.

### How to test it?

**send `GET` request with `locale`**
- ex) `http://localhost:1337/api/homepage?locale=ko`
- `http://localhost:1337/api/homepage?locale=en`
- `http://localhost:1337/api/homepage?locale=all`

### Related issue(s)/PR(s)

Closes #11953

### CODE REVIEW

**Comment by Convly** [comment]
Hi, thank you for opening this PR.

There is indeed an issue between the code only allowing the `locale` filter and the documentation talking about a `_locale` filter.

As we moved from `_locale` to `locale` in V4, I don't think we want to support both formats, thus I don't think your change is necessary. Regardless, the tests are very helpful and it could be cool to keep them.

Would you mind updating your PR to reflect this, please? :slightly_smiling_face:

**Comment by iicdii** [comment]
Sure, I will just keep the test and revert others.

File: packages/plugins/i18n/tests/content-api/content-api.test.e2e.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    data.categories = await builder.sanitizedFixturesFor(categoryModel.singularName, strapi);
+  });
+
+  afterAll(async () => {
+    await strapi.destroy();
+    await builder.cleanup();
+  });
+
+  describe('Test collection type', () => {
+    test('Can filter on default locale', async () => {
```
Comment (Convly) [inline packages/plugins/i18n/tests/content-api/content-api.test.e2e.js:121]:
```suggestion
    test('Filter using the default locale when no parameter is specified', async () => {
```

File: packages/plugins/i18n/tests/content-api/content-api.test.e2e.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+      expect(Array.isArray(body.data)).toBe(true);
+      expect(body.data).toHaveLength(2);
+      expect(body.data).toEqual(
+        expect.arrayContaining(data.categories.map(category => transformToRESTResource(category)))
+      );
+    });
+  });
+
+  describe('Test single type', () => {
+    test('Can filter on default locale', async () => {
```
Comment (Convly) [inline packages/plugins/i18n/tests/content-api/content-api.test.e2e.js:167]: Same as above

**Review by Convly — CHANGES_REQUESTED** [review]
Just some wording adjustment but otherwise it's :100:

**Review by Convly — APPROVED** [review]
LGTM, waiting for the tests to pass, then merging :fire:

**Pull request merged**